whois-query/src/main/java/net/ripe/db/whois/query/pipeline/QueryDecoder.java

```diff
@@ -39,4 +51,27 @@ protected void decode(ChannelHandlerContext ctx, String msg, List<Object> list)
     private boolean isTrusted(final Channel channel) {
         return accessControlListManager.isTrusted(((InetSocketAddress)channel.remoteAddress()).getAddress());
     }
+
+    private String getCharsetName(final Query query){
+        if (!query.isCharsetSpecified()){
+            return StandardCharsets.ISO_8859_1.name();
+        }
+
+        final QueryParser queryParser = new QueryParser(query.toString());
+        final String queryCharset = queryParser.getOptionValue(QueryFlag.CHARSET);
```

Comment:
Use `Charset.forName(x)` to validate the character set name (and catch any exception on an invalid name)

Reply:
Also add tests for character set names. Also find a character set that maps ASCII characters to something else, to see if we support that properly.

Reply:
I can not use Charset.forName(x) there. In the task its says:

Make sure "UTF8", "UTF-8", "LATIN1" and "LATIN-1" are supported (We will allow all of these, even if they are not standard in Java)

So if we pass UTF8 or utf8 I think that Charset will not recognise it

Reply:
Talked via Mattermost, we can handle most of the name just using forName...for latin-1 we will do a wrapper around that